### ISSUE TITLE
"Disable until tab is closed" missing

### ISSUE THREAD

**Issue by PiotrDabrowskey** [web-scrobbler/web-scrobbler#2363]
**Describe the bug**
"Disable until tab is closed" option under right click menu of the extension is missing.

**How to reproduce**
1. Go to any YouTube video
2. Right click on Web Scrobbler icon on the right side of the address bar
3. "Disable YouTube connector" is still visible, while "Disable until tab is closed" is missing.

**Expected behavior**
"Disable until tab is closed" option should be visible next to "Disable YouTube connector".

**Screenshots**
![vlcsnap-2020-04-15-16h00m51s413](https://user-images.githubusercontent.com/9080823/79346154-4e7b0980-7f32-11ea-9886-a6c4218ee61e.png)

**Environment (please complete the following information)**
-   OS: Ubuntu 18.04 bionic
-   Browser: Brave version 1.7.92 Chromium: 80.0.3987.163 (Official Build) (64-bit)
-   Extension version: 2.30.1

**Additional context**
I can provide a debug log if needed.

**Comment by alexesprit** [comment]
Hi! This menu item was removed, because it might be confusing in some cases, as we have a single active icon for all tabs, and context menu should match the icon.

How should we handle a case when you have two music websites opened? Currently the extension displays two menu items: one is for an active website (where music is playing), and one is for a current website:
![image](https://user-images.githubusercontent.com/1119267/79347970-02819200-7f3d-11ea-971c-5f9929acc17c.png)

Do we really need to display "Disable X until A" for both websites?

**Comment by PiotrDabrowskey** [comment]
Ah, ok, fair enough. I'm sad to see one of my favorite feature being dropped. I think I can live with disabling the whole connector.

Though, I must admit, it is a huge step backward for my workflow, as I don't really play more than one video/music at once, so disabling multiple connectors is not a must have for me. Rather than that, I listen to music while working or so and then, during a break, I open "watch later" playlist on YouTube, disable the extension until the tab is closed, freely watch the videos and as soon as I come back to the tab with music (also from YouTube), music is being scrobbled properly. Now, I'd need to remember to activate the connector again, which kinda sucks. So maybe we do need this button?

**Comment by alexesprit** [comment]
@PiotrDabrowskey sorry if I wasn't clear.

I'm not against moving this feature back. Like I said:

> This menu item was removed, **because it might be confusing in some cases, as we have a single active icon for all tabs**, and context menu should match the icon.

So we need to decide when and how is to display "Disable X until A". Is it ok to display two context menu items for both website in the case from my screenshot?

**Comment by PiotrDabrowskey** [comment]
Okay, great then!

> Is it ok to display two context menu items for both website in the case from my screenshot?
I think it works great for two separated websites. However, I do have some doubts regarding the behavior of such option for the same website opened in separated tabs. Supposedly, I have two YouTube tabs opened and one of which I don't want to scrobble from. Currently, only one button for disabling the connector appears, which is fine as both of the tabs are YouTube. Now, would you like to show "Disable X until A" for one or two tabs? If we display two buttons, user won't know which tab the button is referring to, unless we display a tab name - which is not a bad solution but maybe a little too complicated.

Considering this, I'm leaning towards displaying it only for an active tab, even though it might be confusing for some users (because we display "disable connector" option for other tabs, but won't allow this feature to be accessible without activating the tab we want to have scrobbling disabled). Do you think single menu item for an active tab would be feasible? Or maybe have a better idea for the problem I described?

**Comment by alexesprit** [comment]
It's a mess when you open two YouTube tabs and play a video in one of them...

![image](https://user-images.githubusercontent.com/1119267/79386713-324b8c80-7f73-11ea-92ae-177362f135f5.png)

**Comment by PiotrDabrowskey** [comment]
Haha, it is exactly what I was worried about :p

**Comment by alexesprit** [comment]
@PiotrDabrowskey thanks for your comment.

> Considering this, I'm leaning towards displaying it only for an active tab, even though it might be confusing for some users (because we display "disable connector" option for other tabs, but won't allow this feature to be accessible without activating the tab we want to have scrobbling disabled). Do you think single menu item for an active tab would be feasible? Or maybe have a better idea for the problem I described?

I think it's a *probably* best solution we can have.

1. There won't be a situation like on the screenshot I posted.
2. If you press "Disable X until A" for *inactive* tab, there won't be a feedback (icon change), because inactive tabs aren't supposed to change the icon.

On the other side we could hide a second "block" of menu items (I marked them as "These items are for inactive websites"), if both tabs contains the same website. So, there're won't be a mess, like on the screenshot above. The "Disable X until A" menu item for inactive will have no feedback, though.

**Comment by alexesprit** [comment]
@PiotrDabrowskey I have a question regarding these menu items. Is not related to the issue, but it's related to your workflow.

How should we display menu items in a case when two tabs with the same website are opened, **but the current tab is not the active**? It's obvious we should not display menu items for both tabs. So, there're two options:
A. Display menu items for an *active* tab.
B. Display menu items for an *inactive (current)* tab.

What is better?

**Comment by PiotrDabrowskey** [comment]
Thank you for such great engagement in my issue. I really appreciate that :)

> So, there're two options:
> A. Display menu items for an active tab.
> B. Display menu items for an inactive (current) tab.
> 
> What is better?

I guess by active you mean a tab currently playing media and current means a tab visible on the screen, yeah? I think it should apply for a current/visible tab - exactly how it used to be. It's a perfect solution for me, because I can disable scrobbling while being on a "watch later" playlist page, i.e. even before I open any video. In that way, when I open a video to watch, even "now playing" status does not get updated on the last.fm page.

**Comment by alexesprit** [comment]
> I guess by active you mean a tab currently playing media and current means a tab visible on the screen, yeah?

Yes, you're totally right.

> I think it should apply for a current/visible tab - exactly how it used to be. It's a perfect solution for me, because I can disable scrobbling while being on a "watch later" playlist page, i.e. even before I open any video.

So, the current tab is primary, that makes sense. I will also update the order of menu items, so items for a current tab will be added first.

I will try to update the extension and publish a patch update, so it will work in that way.

**Comment by PiotrDabrowskey** [comment]
Great, thanks!

**Comment by alexesprit** [comment]
@PiotrDabrowskey I published v2.30.5, and it should be available shortly. Please check if context menu items work as expected.

**Comment by PiotrDabrowskey** [comment]
Works exactly as expected, thanks! I'm closing this issue.

One thing that came to my mind - "disable until tab is closed" does not disable the whole scrobbler until a certain tab is closed. This used to be confusing when I started using the extension, but I got used to it and completely stopped thinking about it. Anyway, just a side note, maybe more explicit naming, such as "Disable for current tab" would make more sense?
